Microsoft.Identity.Web/ServiceCollectionExtensions.cs

```diff
@@ -38,5 +38,13 @@ public static IServiceCollection AddTokenAcquisition(this IServiceCollection ser
             services.AddScoped<ITokenAcquisition, TokenAcquisition>();
             return services;
         }
+
+        public static IServiceCollection AddTokenAcquisitionB2C(this IServiceCollection services)
```

Comment:
>B2C [](start = 60, length = 3)

we (client side libraries) are trying to move away from B2C specific branding...is there a way to do that here as well? i guess it comes down to what differences are there between B2C and AAD and can we just use the same stuff for each w/minimal changes. should be possible.

Reply:
So, this was actually one of the biggest challenge for me in this sample. Summarizing the challange:

**Problem 1**: On of the main class on Microsoft.Identity.Web is `TokenAcquisition`. There, we use dependency injection for `AzureADOptions`. B2C options on .NET uses the class `AzureADB2COptions` and it is not an exension or related to `AzureADOptions`.

**Problem 2**: The way we build a `IConfidentialClientApplication` is slightly different from AAD and B2C apps. One uses `WithAuthority` while the other uses `WithB2CAuthority`. One has the authority as `{applicationOptions.Instance}{applicationOptions.TenantId}/` while the other is `{azureAdB2COptions.Instance}/tfp/{azureAdB2COptions.Domain}/{azureAdB2COptions.DefaultPolicy}`. 

**Problem 3**: The way we retrieve an `IAccount` is different between the two. AAD will have the `IAccount` identification as `{oid}.{tid}` while B2C would be `{oid}-{tfp}.{tid}`. Furthermore into this problem is B2C not having `tid` claims by default (out of the box), and is a [known issue for us](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/AAD-B2C-specifics#known-issue-with-azure-b2c). It requires the custom claims mapping process to have it on the ID token. So this result in a different approach on retriving the authenticated user `IAccount` on B2C.

So, with all this problems to solve, I couldnt manage to solve it using the same `TokenAcquisition` class, also considering a clean code approach. I will need more help if we decide to have a single class to manage AAD and B2C.

Reply:
yeah...i figured those were the issues. Maybe we can sync w/ @jmprieur on some of these issues as well, we've been tackling and discussing a lot of them. I think we just want MSIdentityWeb to be as "clean" as possible. Esp w/the B2C + AAD merge that will happen, but maybe we just have to go w/something else for the short term? idk, but am up for discussing.